### PR TITLE
Adds categories table and reference/foreign key to category in items table. Adds category model and model testing. Configures shoulda-matchers in rails helper. Sets up has_many/belongs_to relationship between item and category models.

### PR DESCRIPTION
**Pull request by cmacaulay**

### CODE REVIEW

File: app/controllers/items_controller.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class ItemsController < ApplicationController
+
+  def index
+    # byebug
+    @items = Item.all
```
Comment (cmacaulay) [inline app/controllers/items_controller.rb:5]: This was all Charlotte:-)

File: db/migrate/20170222010144_create_categories.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class CreateCategories < ActiveRecord::Migration[5.0]
```
Comment (cmacaulay) [inline db/migrate/20170222010144_create_categories.rb:1]: Remember to migrate! This makes the new table.

File: db/migrate/20170222010254_add_category_reference_to_items.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class AddCategoryReferenceToItems < ActiveRecord::Migration[5.0]
```
Comment (cmacaulay) [inline db/migrate/20170222010254_add_category_reference_to_items.rb:1]: This sets up the relationship!

File: db/schema.rb [hunk 4]
```diff
@@ -10,18 +10,25 @@
...
 
-ActiveRecord::Schema.define(version: 20170221225907) do
+ActiveRecord::Schema.define(version: 20170222010254) do
 
   # These are extensions that must be enabled in order to support this database
   enable_extension "plpgsql"
 
+  create_table "categories", force: :cascade do |t|
+    t.string "name"
+  end
```
Comment (cmacaulay) [inline db/schema.rb:20]: new table.

File: db/schema.rb [hunk 5]
```diff
@@ -10,18 +10,25 @@
...
   create_table "items", force: :cascade do |t|
     t.string   "title"
     t.string   "description"
-    t.decimal  "price",       precision: 2
+    t.float    "price"
     t.string   "image_tag"
     t.datetime "created_at",                null: false
     t.datetime "updated_at",                null: false
+    t.integer  "category_id"
+    t.index ["category_id"], name: "index_items_on_category_id", using: :btree
```
Comment (cmacaulay) [inline db/schema.rb:30]: Does this look good to everyone?

File: spec/models/category_spec.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+require 'rails_helper'
+
```
Comment (cmacaulay) [inline spec/models/category_spec.rb:2]: shoulda-matchers FTW - these tests all pass

File: spec/rails_helper.rb [hunk 7]
```diff
@@ -6,7 +6,12 @@
 require 'spec_helper'
 require 'rspec/rails'
 # Add additional requires below this line. Rails is not loaded until this point!
-
+Shoulda::Matchers.configure do |config|
```
Comment (cmacaulay) [inline spec/rails_helper.rb:9]: This configures shoulda-matchers.

**Review by cmacaulay — COMMENTED** [review]
@CjMoore @valerietrudell @meyerhoferc ready for review!

File: db/schema.rb (hunk 5 above)
Comment (meyerhoferc) [reply db/schema.rb:30]: glad we have the index for the schema

**Comment by meyerhoferc** [comment]
I'm good with this merge as long as we talk about how to deal with the HOUND sometime

**Comment by valerietrudell** [comment]
Yeah, there are a lot of hound comments here but it sounds like they will all come up again when we push to master again. Looks good though.

**Pull request merged**